index/cache.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2023-2024 Princess Beef Heavy Industries, LLC / Dave Shanley
+// https://pb33f.io
+
+package index
+
+import "sync"
+
+func (index *SpecIndex) SetCache(sync *sync.Map) {
+	index.cache = sync
+}
+
+// HighCacheHit increments the counter of high cache hits by one, and returns the current value of hits.
+func (index *SpecIndex) HighCacheHit() uint64 {
+	index.highModelCacheHits.Add(1)
+	return index.highModelCacheHits.Load()
+}
```

Comment:
I would suggest you create a custom type that is a wrapper around the sync.Map and expose setter/getter methods. This way the caller doesn't have to remember to call the CacheHit/Miss functions and your metrics will be much more reliable.